doc/handle-ASIC-SDK-health-event/handle-ASIC-SDK-health-event.md

```diff
@@ -0,0 +1,635 @@
+
+``` C
+/**
+ * @brief Switch health event severity
+ */
+typedef enum _sai_switch_asic_sdk_health_severity_t
+{
+    /** Switch event severity fatal */
+    SAI_SWITCH_ASIC_SDK_HEALTH_SEVERITY_FATAL,
+
+    /** Switch event severity warning */
+    SAI_SWITCH_ASIC_SDK_HEALTH_SEVERITY_WARNING,
+
+    /** Switch event severity notice */
+    SAI_SWITCH_ASIC_SDK_HEALTH_SEVERITY_NOTICE
+
+} sai_switch_asic_sdk_health_severity_t;
+
+/**
+ * @brief Switch health categories
+ */
+typedef enum _sai_switch_asic_sdk_health_category_t
+{
+    /** Switch health software category */
+    SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_SW,
+
+    /** Switch health firmware category */
+    SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_FW,
+
+    /** Switch health cpu hardware category */
+    SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_CPU_HW,
+
+    /** Switch health ASIC hardware category */
+    SAI_SWITCH_ASIC_SDK_HEALTH_CATEGORY_ASIC_HW
+
+} sai_switch_asic_sdk_health_category_t;
+
+/**
+ * @brief Switch health event callback
+ *
+ * @objects switch_id SAI_OBJECT_TYPE_SWITCH
+ *
+ * @param[in] switch_id Switch Id
+ * @param[in] severity Health event severity
+ * @param[in] timestamp Time and date of receiving the SDK Health event
+ * @param[in] category Category of cause
+ * @param[in] data Data of switch health
+ * @param[in] description JSON-encoded description string with information delivered from SDK event/trap
+ * Example of a possible description:
+ * {
+ *    "switch_id": "0x00000000000000AB",
+ *    "severity": "2",
+ *    "timestamp": {
+ *        "tv_sec": "22429",
+ *        "tv_nsec": "3428724"
+ *    },
+ *    "category": "3",
+ *    "data": {
+ *        data_type: "0"
+ *    },
+ *    "additional_data": "Some additional information"
+ * }
+ */
+
+typedef void (*sai_switch_asic_sdk_health_event_notification_fn)(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_switch_asic_sdk_health_severity_t severity,
+        _In_ sai_timespec_t timestamp,
+        _In_ sai_switch_asic_sdk_health_category_t category,
+        _In_ sai_switch_health_data_t data,
+        _In_ const sai_u8_list_t description);
+
+```
+
+The following type definitions for the SAI attributes defined in `SAI/inc/saitypes.h` are used in this document.
+
+```C
+typedef enum _sai_health_data_type_t
+{
+    /** General health data type */
+    SAI_HEALTH_DATA_TYPE_GENERAL
+} sai_health_data_type_t;
+
+typedef struct _sai_switch_health_data_t
+{
+    /** Type of switch health data */
+    sai_health_data_type_t data_type;
+
+} sai_switch_health_data_t;
+```
+
+### Configuration and management
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A.
+
+#### CLI Enhancements
+
+##### Configure suppress ASIC/SDK health events by severity and category
+
+Command `config asic-sdk-health-event suppress <severity> <category-list>|<none>|<all>` is introduced for a customer to configure the categories that he/she wants to suppress for a certain severity.
```

Comment:
How does it work for multi-asic platforms?

Reply:
There is a `--namespace` option. By default, it's for all ASICs.